src/databricks/labs/ucx/mixins/fixtures.py

```diff
@@ -638,6 +639,33 @@ def _scim_values(ids: list[str]) -> list[iam.ComplexValue]:
     return [iam.ComplexValue(value=x) for x in ids]
 
 
+def wait_group_provisioned(interface: AccountGroupsAPI | GroupsAPI, *groups: Group) -> None:
+    # The groups API is eventually consistent, but not monotonically consistent. Here we try to compensate for
+    # the lack of monotonic consistency by requiring that two subsequent calls confirm it exists. REST API
+    # internals cache things for up to 60s, we see times close to this during tests. The retry timeout reflects
+    # this: if it's taking much longer then something else is wrong.
+
+    @retried(on=[NotFound], timeout=timedelta(seconds=90))
```

Comment:
please also make the relevant PR into https://github.com/databrickslabs/pytester